issuance/cert.go

```diff
@@ -47,6 +52,7 @@ type PolicyConfig struct {
 
 // Profile is the validated structure created by reading in ProfileConfigs and IssuerConfigs
 type Profile struct {
+	Name            string
```

Comment:
Similar to my comment in //cmd/boulder-ca/main.go, I don't think it's necessary for Profile to contain a name or a hash. Specifically, I don't think the issuance package cares about the name or the hash at all: its job is to produce a certificate from the inputs, and the certificate won't contain either the name or the hash of the profile, so those don't need to be in the input at all.

Instead, the name and the hash can be managed entirely inside the //ca package, as side-by-side map[name]Profile and map[hash]Profile, which you're already doing.

This also removes the need for GetHash and UnsafeSetHash.